packages/roosterjs-content-model-plugins/lib/edit/deleteSteps/deleteList.ts

```diff
@@ -1,22 +1,32 @@
 import { getClosestAncestorBlockGroupIndex } from 'roosterjs-content-model-dom';
-import type { DeleteSelectionStep } from 'roosterjs-content-model-types';
+import type { DeleteSelectionStep, ContentModelListItem } from 'roosterjs-content-model-types';
 
 /**
  * @internal
  */
 export const deleteList: DeleteSelectionStep = context => {
+    if (context.deleteResult != 'notDeleted') {
+        return;
+    }
+
     const { paragraph, marker, path } = context.insertPoint;
 
-    if (context.deleteResult == 'nothingToDelete' || context.deleteResult == 'notDeleted') {
-        const index = getClosestAncestorBlockGroupIndex(path, ['ListItem', 'TableCell']);
-        const item = path[index];
-        if (
-            item &&
-            index >= 0 &&
-            paragraph.segments[0] == marker &&
-            item.blockGroupType == 'ListItem'
-        ) {
-            item.levels = [];
+    if (paragraph.segments[0] == marker) {
+        const index = getClosestAncestorBlockGroupIndex<ContentModelListItem>(
+            path,
+            ['ListItem'],
+            ['TableCell', 'FormatContainer']
+        );
+        const item = path[index] as ContentModelListItem | undefined;
```

Comment:
Is possible to be other thing than a list?

Reply:
From line 17, we only query ListItem. so here it should be either ListItem or undefined (then index is -1)